File: backend/app/api/v1/routes/notifications.py
from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.notification import NotificationSubscriptionCreateRequest, NotificationSubscriptionResponse
from app.services.notification_service import NotificationService

router = APIRouter()


@router.post("/subscriptions", response_model=NotificationSubscriptionResponse, summary="Create notification subscription")
def create_subscription(payload: NotificationSubscriptionCreateRequest, db: Session = Depends(get_db)):
    return NotificationService(db).create_subscription(payload)


@router.get("/subscriptions", response_model=list[NotificationSubscriptionResponse], summary="List notification subscriptions")
def list_subscriptions(db: Session = Depends(get_db)):
    return NotificationService(db).list_subscriptions()

File: frontend/src/api/notifications.js
import apiClient from './client';

export async function listSubscriptions() {
  const { data } = await apiClient.get('/notifications/subscriptions');
  return data;
}

export async function createSubscription(payload) {
  const { data } = await apiClient.post('/notifications/subscriptions', payload);
  return data;
}
